components/org.wso2.carbon.identity.api.server.script.library/org.wso2.carbon.identity.api.server.script.library.v1/src/main/java/org/wso2/carbon/identity/api/server/script/library/v1/core/ServerScriptLibrariesService.java

```diff
@@ -0,0 +1,439 @@
+import org.wso2.carbon.identity.api.server.script.library.common.Constants;
+import org.wso2.carbon.identity.api.server.script.library.common.ScriptLibraryServiceHolder;
+import org.wso2.carbon.identity.api.server.script.library.v1.model.ScriptLibrary;
+import org.wso2.carbon.identity.api.server.script.library.v1.model.ScriptLibraryListResponse;
+import org.wso2.carbon.identity.api.server.script.library.v1.model.ScriptLibraryPOSTRequest;
+import org.wso2.carbon.identity.api.server.script.library.v1.model.ScriptLibraryPUTRequest;
+import org.wso2.carbon.identity.api.server.script.library.v1.model.ScriptLibraryResponse;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.functions.library.mgt.exception.FunctionLibraryManagementClientException;
+import org.wso2.carbon.identity.functions.library.mgt.exception.FunctionLibraryManagementException;
+import org.wso2.carbon.identity.functions.library.mgt.exception.FunctionLibraryManagementServerException;
+import org.wso2.carbon.identity.functions.library.mgt.model.FunctionLibrary;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.common.Constants.V1_API_PATH_COMPONENT;
+import static org.wso2.carbon.identity.api.server.script.library.common.Constants.SCRIPT_LIBRARY_CONTENT_PATH;
+import static org.wso2.carbon.identity.api.server.script.library.common.Constants.SCRIPT_LIBRARY_PATH_COMPONENT;
+
+/**
+ * Call internal osgi services to perform server script library management operations.
+ */
+public class ServerScriptLibrariesService {
+
+    private static final Log log = LogFactory.getLog(ServerScriptLibrariesService.class);
+    private static final int DEFAULT_OFFSET = 0;
+
+    /**
+     * Get list of Script Libraries.
+     *
+     * @param limit  Items per page.
+     * @param offset Offset.
+     * @return ScriptLibraryListResponse.
+     */
+    public ScriptLibraryListResponse getScriptLibraries(Integer limit, Integer offset) {
+
+        List<FunctionLibrary> functionLibraries;
+        try {
+            functionLibraries = ScriptLibraryServiceHolder.getScriptLibraryManagementService()
+                    .listFunctionLibraries(ContextLoader.getTenantDomainFromContext());
+        } catch (FunctionLibraryManagementException e) {
+            throw handleScriptLibraryError(e, Constants.ErrorMessage.ERROR_CODE_ERROR_LISTING_SCRIPT_LIBRARIES);
+        }
+        return createScriptLibrariesList(functionLibraries, validateLimit(limit), validateOffset(offset));
+    }
+
+    /**
+     * Validate the offset.
+     *
+     * @param offset Offset value
+     * @return Validated offset.
+     */
+    private int validateOffset(Integer offset) {
+
+        if (offset != null && offset >= 0) {
```

Comment:
let's throw an error (Bad request ) if offset is negative. Return default value for null offset

Reply:
[Fixed commit](https://github.com/wso2/identity-api-server/pull/81/commits/f1f24a495063022b48d0fe2a91be8b83c6cf07de)